### PR TITLE
Modifications needed for new mapl generic

### PR DESCRIPTION
**Pull request by tclune**
These are some changes that are necessary to keep up with some refactoring in MAPL.   I think these can be approved as is, but someone should check that the existing model builds fine with these changes (and without the MAPL changes).

### CODE REVIEW

**Comment by tclune** [comment]
I have verified that GCM builds with 0-diffs with this branch.

**Comment by mathomp4** [comment]
I suppose we should have @sanAkel make sure this looks good to him. These codes I mostly associate with him.

File: GEOS_Util/pre/NSIDC-OSTIA_SST-ICE_blend/regrid_forcing.F90 [hunk 1]
```diff
@@ -614,7 +614,6 @@ Subroutine do_regrid_forcing(rc)
   type(ESMF_Config)            :: cf_root
   type(ESMF_Config)            :: cf_hist
   character(len=ESMF_MAXSTR), parameter :: CF_FILE='REGRID_FORCING.rc'
-  type(MAPL_Communicators) :: mapl_comm
```
Comment (sanAkel) [inline GEOS_Util/pre/NSIDC-OSTIA_SST-ICE_blend/regrid_forcing.F90:617]: I would have to refer this change to @bena-nasa he wrote this subroutine

File: GEOS_Util/pre/NSIDC-OSTIA_SST-ICE_blend/regrid_forcing_esmf.F90 [hunk 2]
```diff
@@ -579,7 +579,7 @@ Subroutine do_regrid_forcing(rc)
   type(ESMF_Config)            :: cf_root
   type(ESMF_Config)            :: cf_hist
   character(len=ESMF_MAXSTR), parameter :: CF_FILE='REGRID_FORCING.rc'
-  type(MAPL_Communicators) :: mapl_comm
```
Comment (sanAkel) [inline GEOS_Util/pre/NSIDC-OSTIA_SST-ICE_blend/regrid_forcing_esmf.F90:582]: I would have to refer this change to @bena-nasa he wrote this subroutine

File: GEOS_Util/pre/NSIDC-OSTIA_SST-ICE_blend/regrid_forcing_esmf.F90 [hunk 3]
```diff
@@ -603,9 +603,6 @@ Subroutine do_regrid_forcing(rc)
   VERIFY_(STATUS)
 
 
-  mapl_comm%esmf%comm=MPI_COMM_WORLD
```
Comment (sanAkel) [inline GEOS_Util/pre/NSIDC-OSTIA_SST-ICE_blend/regrid_forcing_esmf.F90:606]: I would have to refer this change to @bena-nasa he wrote this subroutine

Comment (sanAkel) [reply GEOS_Util/pre/NSIDC-OSTIA_SST-ICE_blend/regrid_forcing_esmf.F90:606]: do you guys (@tclune @mathomp4 ) want me to _test_ these changes? FWIK I may have to check with `OPS` if they are at all planning to use the code from _here_ (git)!

**Comment by tclune** [comment]
@sanAkel It would be useful to verify that my changes do not break anything.    By eye, they really should be safe as the deletion should not even compile if the fragments were needed by anything.

So if the testing is easy.   Or even tell @mathomp4 and myself how to verify the executables still work.

**Comment by sanAkel** [comment]
> @sanAkel It would be useful to verify that my changes do not break anything. By eye, they really should be safe as the deletion should not even compile if the fragments were needed by anything.
> 
> So if the testing is easy. Or even tell @mathomp4 and myself how to verify the executables still work.

Unfortunately I don't have a _recent_ build. Let me reach @mathomp4 over teams ...

**Comment by mathomp4** [comment]
I'm putting back the blocking label on this. @bena-nasa did some tests and it looks like this PR breaks the programs with or without the MAPL update from @tclune (https://github.com/GEOS-ESM/MAPL/pull/677). @bena-nasa can comment more on his tests but I think his finding was v10.17.1 it works, v10.17.1 + this PR doesn't, and neither does v10.17.1 + #155 + https://github.com/GEOS-ESM/MAPL/pull/677

I know @sdrabenh wants to make a release of GMAO_Shared with the plots and other updates here. If we can figure out a fix for this (in MAPL or here). We could make a quick zero-diff patch release to GMAO_Shared to let @tclune continue his work in MAPL.

**Comment by mathomp4** [comment]
This definitely works with https://github.com/GEOS-ESM/MAPL/pull/677 now. So it's good to go.

**Review by sdrabenh — APPROVED** [review]

**Pull request merged**